source/extensions/upstreams/tcp/generic/config.cc

```diff
@@ -16,10 +16,15 @@ TcpProxy::GenericConnPoolPtr GenericConnPoolFactory::createGenericConnPool(
     const absl::optional<TunnelingConfig>& config, Upstream::LoadBalancerContext* context,
     Envoy::Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks) const {
   if (config.has_value()) {
-    auto pool_type =
-        ((thread_local_cluster.info()->features() & Upstream::ClusterInfo::Features::HTTP2) != 0)
-            ? Http::CodecType::HTTP2
-            : Http::CodecType::HTTP1;
+    Http::CodecType pool_type;
+    if ((thread_local_cluster.info()->features() & Upstream::ClusterInfo::Features::HTTP3) != 0) {
```

Comment:
Hm.  I wonder if we should check for and reject config with H3 and H3 present for now.  Eventually we may want to allow auto_http semantics for this path but until we do I'd prefer we force reject "invalid" config.

Reply:
Where would you suggest performing that check? I suspect we'd want it as close as possible to where the config is parsed

Reply:
Yeah, thinking more this may not be possible given clusters can be used for L4+connect or normal HTPT proxying.  In that case I think until we support ALPN I'd do H2 over H3, so if for some reason there's mixed traffic and we're not going to do ALPN we basically always auto-fall-back rather than sometimes black holing.